zap/src/main/java/org/zaproxy/zap/view/table/DefaultHistoryReferencesTableEntry.java

```diff
@@ -120,6 +129,52 @@ public DefaultHistoryReferencesTableEntry(HistoryReference historyReference, Col
 
         alertRiskCellItem = super.getHighestAlert();
 
+        try {
+            hostname1 =
+                    hasColumn(sortedColumns, Column.HOSTNAME)
+                            ? historyReference.getURI().getHost()
+                            : null;
+        } catch (URIException e) {
+            hostname1 = null;
+            e.printStackTrace();
+        }
+        hostname = hostname1;
```

Comment:
Since these are basically all the same why not extract a private method and pass the column constant?
Printing the stack trace should be replaced with a logger.

Reply:
Also is the intermediate storage necessary? (var vs var1)

Reply:
That's because they are final instance variables. Although IMO the components should be shown encoded, showing them decoded leads to data loss and misleading values (crude example, the path `/%2F` will be shown as `//` which is a completely different path).

Reply:
@thc202 I also think that they should be shown encoded, I didn't notice that was happening. @kingthorin the functions I will use now don't throw exception and the code will become more clean.

Reply:
The host should also be shown encoded (for same reason), e.g.:
`new String(historyReference.getURI().getRawHost())`